src/careamics/config/data/ng_data_config.py

```diff
@@ -136,6 +136,12 @@ class NGDataConfig(BaseModel):
     batch_size: int = Field(default=1, ge=1, validate_default=True)
     """Batch size for training."""
 
+    in_memory: bool = Field(default=None, validate_default=True)
```

Comment:
Could use Pydantic's [default factory option](https://docs.pydantic.dev/latest/concepts/fields/#default-values), I would maybe prefer it to having a "before" validator that modifies the value, because it is less hidden.


```suggestion
    in_memory: bool = Field(
        default_factory=lambda data: (
            False if data["data_type"] in ["zarr", "czi"] else True
        ),
        validate_default=True,
    )
```

Reply:
Good idea!

I missed you commit suggestion, I already had pushed my commit.